src/torcontrol.cpp

```diff
@@ -132,29 +132,36 @@ void TorControlConnection::eventcb(struct bufferevent *bev, short what, void *ct
 
 bool TorControlConnection::Connect(const std::string& tor_control_center, const ConnectionCB& _connected, const ConnectionCB& _disconnected)
 {
-    if (b_conn)
+    if (b_conn) {
         Disconnect();
-    // Parse tor_control_center address:port
-    struct sockaddr_storage connect_to_addr;
-    int connect_to_addrlen = sizeof(connect_to_addr);
-    if (evutil_parse_sockaddr_port(tor_control_center.c_str(),
-        (struct sockaddr*)&connect_to_addr, &connect_to_addrlen)<0) {
-        LogPrintf("tor: Error parsing socket address %s\n", tor_control_center);
+    }
+
+    CService control_service;
+    if (!Lookup(tor_control_center, control_service, 9051, fNameLookup)) {
+        LogPrintf("tor: Failed to look up control center %s\n", tor_control_center.c_str());
+        return false;
+    }
+
+    struct sockaddr_storage control_address;
+    socklen_t control_address_len = sizeof(sockaddr);
+    if (!control_service.GetSockAddr(reinterpret_cast<struct sockaddr*>(&control_address), &control_address_len)) {
+        LogPrintf("tor: Error parsing socket address %s\n", tor_control_center.c_str());
         return false;
     }
 
     // Create a new socket, set up callbacks and enable notification bits
     b_conn = bufferevent_socket_new(base, -1, BEV_OPT_CLOSE_ON_FREE);
-    if (!b_conn)
+    if (!b_conn) {
         return false;
+    }
     bufferevent_setcb(b_conn, TorControlConnection::readcb, nullptr, TorControlConnection::eventcb, this);
     bufferevent_enable(b_conn, EV_READ|EV_WRITE);
     this->connected = _connected;
     this->disconnected = _disconnected;
 
     // Finally, connect to tor_control_center
-    if (bufferevent_socket_connect(b_conn, (struct sockaddr*)&connect_to_addr, connect_to_addrlen) < 0) {
-        LogPrintf("tor: Error connecting to address %s\n", tor_control_center);
+    if (bufferevent_socket_connect(b_conn, reinterpret_cast<struct sockaddr*>(&control_address), control_address_len) < 0) {
+        LogPrintf("tor: Error connecting to address %s\n", tor_control_center.c_str());
```

Comment:
Untested, but ISTM you can omit the 3 uses of `c_str()`

<details><summary>diff</summary><p>

```diff
diff --git a/src/torcontrol.cpp b/src/torcontrol.cpp
index 4a56159228..dcb2175c94 100644
--- a/src/torcontrol.cpp
+++ b/src/torcontrol.cpp
@@ -138,14 +138,14 @@ bool TorControlConnection::Connect(const std::string& tor_control_center, const
 
     CService control_service;
     if (!Lookup(tor_control_center, control_service, 9051, fNameLookup)) {
-        LogPrintf("tor: Failed to look up control center %s\n", tor_control_center.c_str());
+        LogPrintf("tor: Failed to look up control center %s\n", tor_control_center);
         return false;
     }
 
     struct sockaddr_storage control_address;
     socklen_t control_address_len = sizeof(sockaddr);
     if (!control_service.GetSockAddr(reinterpret_cast<struct sockaddr*>(&control_address), &control_address_len)) {
-        LogPrintf("tor: Error parsing socket address %s\n", tor_control_center.c_str());
+        LogPrintf("tor: Error parsing socket address %s\n", tor_control_center);
         return false;
     }
 
@@ -161,7 +161,7 @@ bool TorControlConnection::Connect(const std::string& tor_control_center, const
 
     // Finally, connect to tor_control_center
     if (bufferevent_socket_connect(b_conn, reinterpret_cast<struct sockaddr*>(&control_address), control_address_len) < 0) {
-        LogPrintf("tor: Error connecting to address %s\n", tor_control_center.c_str());
+        LogPrintf("tor: Error connecting to address %s\n", tor_control_center);
         return false;
     }
```
</p></details>

(Optional, you'll save time by squashing the commits after each update, to the final version for review.)

Reply:
Indeed, I see now `LogPrintf` this is based on templates, not untyped varargs.